crates/wysiwyg/src/dom/find_range.rs

```diff
@@ -149,7 +149,10 @@ where
     let container_end = *offset;
     let container_node_len = container_end - container_start;
     // We never want to return the root node
-    if !node.handle().is_root() {
+    if container_end >= start
+        && container_start <= end
+        && !node.handle().is_root()
+    {
```

Comment:
This is not really needed for the PR, but I just realised we were always adding *any* container node as part of the selection. It didn't matter much in general because we always checked the `is_leaf` nodes first, but this is more correct.